core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -0,0 +1,346 @@
+package edu.asu.c3simulator.widgets;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.badlogic.gdx.Game;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+/**
+ * This class creates a navigation panel by creating the buttons and linking them with
+ * their screens. It also supports having one level of sub-buttons to a parent button. A
+ * parent button will only show and hide sub-buttons when clicked. Only one of the parent
+ * buttons can show its sub-buttons at the same time. If a parent button is clicked, the
+ * already extended sub-buttons will collapse before extending the new set of sub-buttons.
+ * 
+ * @author Alyahya, Mohammed
+ */
+public class NavigationPanel extends Table
+{
+	private class NavigationButtonListener extends ClickListener
+	{
+		String screenName;
+		Screen targetedScreen;
+		
+		public NavigationButtonListener(String screenName, Screen targetedScreen)
+		{
+			this.screenName = screenName;
+			this.targetedScreen = targetedScreen;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			System.out.println(screenName);
+			game.setScreen(targetedScreen);
+		}
+	}
+	
+	private class CollapsibleButtonListener extends ClickListener
+	{
+		String screenName;
+		
+		public CollapsibleButtonListener(String screenName)
+		{
+			this.screenName = screenName;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			if (displayedCollapsibleButton.equalsIgnoreCase(screenName))
+				displayedCollapsibleButton = "";
+			else
+				displayedCollapsibleButton = screenName;
+			recreateNavigationPanel();
+		}
+	}
+	
+	private Game game;
+	private Skin skin;
+	private LinkedHashMap<String, Screen> buttons;
+	private LinkedHashMap<String, LinkedHashMap<String, Screen>> subButtons;
```

Comment:
Flagged for refactoring

ButtonListeners might be a better choice, and it could eliminate these hash maps.
The nested maps are a bit brutal